File: frontend/user/lib/data/mapper/auth/user_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/models/auth/user_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/auth/user.dart';

class UserMapper {
  static User fromModel(UserModel model) {
    return User(
      id: model.id,
      email: model.email,
      username: model.username,
      role: model.role,
    );
  }

  static Map<String, dynamic> toJson(User entity) {
    return {
      "id": entity.id,
      "email": entity.email,
      "username": entity.username,
      "role": entity.role,
    };
  }
}

File: backend/app/infrastructure/database/mappers/user_mapper.py
from app.domain.entities.user_entity import UserEntity
from app.infrastructure.database.models.user import User


class UserMapper:
    @staticmethod
    def to_entity(model: User) -> UserEntity:
        return UserEntity(
            id=model.id,
            email=model.email,
            username=model.username,
            password=model.password,
            role=model.role.value if hasattr(model.role, 'value') else model.role,
            created_at=model.created_at,
            updated_at=model.updated_at,
        )

    @staticmethod
    def to_model(entity: UserEntity) -> User:
        return User(
            id=entity.id,
            email=entity.email,
            username=entity.username,
            password=entity.password,
            role=entity.role,
        )
